src/main/java/io/appium/java_client/remote/DirectConnect.java

```diff
@@ -0,0 +1,73 @@
+package io.appium.java_client.remote;
+
+import javax.annotation.Nullable;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static io.appium.java_client.internal.CapabilityHelpers.APPIUM_PREFIX;
+
+// TODO: simplify with lombok as another PR
+public class DirectConnect {
+    private static final String DIRECT_CONNECT_PROTOCOL = "directConnectProtocol";
+    private static final String DIRECT_CONNECT_PATH = "directConnectPath";
+    private static final String DIRECT_CONNECT_HOST = "directConnectHost";
+    private static final String DIRECT_CONNECT_PORT = "directConnectPort";
+
+    public final String protocol;
```

Comment:
please create public accessors rather than making properties public (e.g. protocol -> getProtocol()). Later we could simplify that code with lombok